ui/ducks/confirm/confirm.ts

```diff
@@ -1,11 +1,18 @@
-type ActionType = {
-  type: string;
-  currentConfirmation?: Record<string, unknown> | undefined;
-};
+type ActionType =
+  | {
+      type: 'metamask/confirm/UPDATE_CONFIRM';
+      payload?: Record<string, unknown>;
+    }
+  | {
+      type: 'metamask/confirm/UPDATE_CURRENT_CONFIRMATION';
+      currentConfirmation?: Record<string, unknown> | undefined;
+    };
 
-const createActionType = (action: string): string =>
-  `metamask/confirm/${action}`;
+const createActionType = <T extends string>(
+  action: T,
+): `metamask/confirm/${T}` => `metamask/confirm/${action}`;
```

Comment:
Action will always be string only I think. Not sure if we need generics above.

Reply:
👍🏼  updated/reverted ed39f463e33648bc17125c923ec72e4b314b24b4